dev/scripts/build.ts

```diff
@@ -3,46 +3,139 @@ import { join } from "node:path"
 import { readJson, writeJson } from "../attest/src/fs.js"
 import { shell } from "../attest/src/shell.js"
 import { repoDirs } from "./common.js"
+import { rewritePaths } from "./overwrite.js"
+import type { ReplacementDictionary } from "./overwrite.js"
 
 const packageRoot = process.cwd()
 const outRoot = join(packageRoot, "dist")
 const packageJson = readJson(join(packageRoot, "package.json"))
+const tempTsConfigBaseName = "tsconfig.temp"
+const tempTsConfigPath = join(packageRoot, `${tempTsConfigBaseName}.json`)
 
-const buildFormat = (module: "commonjs" | "esnext") => {
-    const outDir = join(outRoot, module === "commonjs" ? "cjs" : "mjs")
-    writeJson(tempTsConfig, {
-        ...tsConfigData,
-        include: ["src"],
+const nuke = (target: string) =>
+    rmSync(target, { recursive: true, force: true })
+const clone = (from: string, to: string): void =>
+    cpSync(from, to, { recursive: true, force: true })
+
+const writeManifest =
+    (overrides: Record<string, unknown>) =>
+    (sourceDir: string, targetDir: string) => {
+        const manifest = readJson(join(sourceDir, "package.json"))
+        writeJson(join(targetDir, "package.json"), {
+            ...manifest,
+            ...overrides
+        })
+    }
+
+const Sources = {
+    utils: ["dev", "utils"],
+    attest: ["dev", "attest"]
+} as const
+
+const replacementDictionary: ReplacementDictionary = {
+    attest: {
+        pattern: /"(\.\.\/)+[^"]*attest\/[^"]*\.js"/g,
+        replacement: `"@arktype/attest"`
+    },
+    utils: {
+        pattern: /"(\.\.\/)+[^"]*utils\/[^"]*\.js"/g,
+        replacement: `"@arktype/utils"`
+    }
+} as const
+
+const ignorePaths = [
+    "node_modules",
+    "package.json",
+    `${tempTsConfigBaseName}.tsbuildinfo`
```

Comment:
Mentioning this for posterity:

The reason I'm deriving this is to make sure that the `.tsbuildinfo` file continues to be ignored even if we change the name of the temporary tsconfig file.

Frankly, this is one of the shortcomings of this approach, since IMO it's hard to reason about how rewriting these paths will impact things like incremental builds, etc.

In this case I don't think it's an issue since the whole reason we did it this way was to avoid adding a build step during local development (so an impact on incremental builds is a moot point).

So it shouldn't be an issue currently, but was worth mentioning anyway.